src/jvm/benchmark/org/jetbrains/kotlinx/lincheck_benchmark/running/ConcurrentDequeBenchmark.kt

```diff
@@ -0,0 +1,32 @@
+/*
+ * Lincheck
+ *
+ * Copyright (C) 2019 - 2023 JetBrains s.r.o.
+ *
+ * This Source Code Form is subject to the terms of the
+ * Mozilla Public License, v. 2.0. If a copy of the MPL was not distributed
+ * with this file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package org.jetbrains.kotlinx.lincheck_benchmark.running
+
+import org.jetbrains.kotlinx.lincheck.annotations.*
+import org.jetbrains.kotlinx.lincheck.paramgen.IntGen
+import org.jetbrains.kotlinx.lincheck_benchmark.AbstractLincheckBenchmark
+import java.util.concurrent.*
+
+class ConcurrentDequeBenchmark : AbstractLincheckBenchmark() {
```

Comment:
It should be similar to `ConcurrentLinkedQueue`, so it might be safe to remove it.

Reply:
What do you mean by "similar" here? I thought those two are two data structures with different implementations. Also, in Lincheck tests we currently have tests for both, that's another reason why I create two different benchmarks for them here.

Reply:
The underlying data structure is pretty similar. The main reason we have both `ConcurrentLinkedQueue` and `ConcurrentLinkedDeque` is that the second one is non-linearizable

Reply:
However, if you find it important to test the linearizable part of it in the benchmarks -- let's keep it. 

By the way, do you want to include incorrect algorithms in the benchmark suit to benchmark how fast Lincheck detects bugs?

Reply:
> By the way, do you want to include incorrect algorithms in the benchmark suit to benchmark how fast Lincheck detects bugs?

Yes, I was planning it. 
But I was planning to add it later in a separate PR, and in a separate category/subfolder of benchmarks of two reasons:

1. To first review and merge this PR with basic benchmarks infrastructure. Once the infrastructure is done, the rest of the work on benchmarks can be parallelized and implemented independently. Also, we will likely have a lot of discussion around particular choice of benchmarks, that would slow down the other parts. 

2. For the incorrect algorithms we would collect different set of metrics, compared to the correct ones. 
For correct we want to collect: overall runtime, coverage, etc. 
For incorrect we want to collect metrics such as: time-to-first-bug, amount of bugs detected in a time unit, etc.